src/main/java/uk/gov/hmcts/reform/pdf/service/aspect/AuthorisationAspect.java

```diff
@@ -0,0 +1,49 @@
+package uk.gov.hmcts.reform.pdf.service.aspect;
+
+import org.aspectj.lang.JoinPoint;
+import org.aspectj.lang.annotation.Aspect;
+import org.aspectj.lang.annotation.Before;
+import org.aspectj.lang.reflect.MethodSignature;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import org.springframework.web.bind.annotation.RequestHeader;
+import uk.gov.hmcts.reform.pdf.service.service.AuthorisationService;
+
+import java.lang.annotation.Annotation;
+
+@Aspect
+@Component
+public class AuthorisationAspect {
+
+    private boolean stop;
```

Comment:
This component will be shared across all requests, using a field to record intermediate call state will render it non-thread safe!